kanidmd/core/src/actors/v1_write.rs

```diff
@@ -346,7 +346,7 @@ impl QueryServerWriteV1 {
         filter: Filter<FilterInvalid>,
         eventid: Uuid,
     ) -> Result<(), OperationError> {
-        let idms_prox_write = self.idms.proxy_write_async(duration_from_epoch_now()).await;
+        let mut idms_prox_write = self.idms.proxy_write(duration_from_epoch_now()).await;
```

Comment:
When an operation (event) starts, we need the "current time" to be identical throughout that event. So in some cases we have the current time only needed being to create the transaction itself (such as the above) and no other elements of the event take time. When current time is needed in multiple calls within the fn, we set ct to a value and re-use it throughout the operation.

A possible improvement here would actually be that when we start the txn we set ct, but then inside of the server, we access ct via the txn instead of from function args. That would guarantee that it's the "current time of the transaction" rather than relying on this "pattern" that currently exists.

So yeah, tl;dr - in any "event" we assume time is a fixed moment (instant) rather than calling to a clock source

The reason generally is that this lets us control exactly what "time" it is during tests so we can advance and rewind time as needed to test edge cases.

Hope that helps :)

Reply:
I think I'm *more* confused now, but if it's for good reason, then so be it 😃